rest/changes_api_test.go

```diff
@@ -3046,6 +3046,46 @@ func TestChangesAdminChannelGrantLongpollNotify(t *testing.T) {
 
 }
 
+func TestTombstoneCompaction(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)
+
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Walrus does not support Xattrs")
+	}
+
+	if !base.TestUseXattrs() {
+		t.Skip("If running with no xattrs compact acts as a no-op")
+	}
+
+	rt := NewRestTester(t, nil)
+
+	count := 0
+
+	rt.GetDatabase().PurgeInterval = 0
+
+	for count < 200 {
+		count++
+		response := rt.SendAdminRequest("POST", "/db/", `{"foo":"bar"}`)
+		assert.Equal(t, 200, response.Code)
+		var body db.Body
+		err := json.Unmarshal(response.Body.Bytes(), &body)
+		assert.NoError(t, err)
+		revId := body["rev"].(string)
+		docId := body["id"].(string)
+
+		response = rt.SendAdminRequest("DELETE", fmt.Sprintf("/db/%s?rev=%s", docId, revId), "")
+		assert.Equal(t, 200, response.Code)
+	}
+
+	rt.SendAdminRequest("POST", "/db/_compact", "")
+
+	assert.Equal(t, int64(200), base.ExpvarVar2Int(rt.GetDatabase().DbStats.StatsDatabase().Get(base.StatKeyNumTombstonesCompacted)))
```

Comment:
A few variations of total docs count would be nice here (a multiple of batch size, less than batch size, etc)